index.js

```diff
@@ -2127,6 +2077,74 @@
                    foldable);
   }
 
+  //# takeWhile :: (Applicative f, Foldable f, Monoid (f a)) => (a -> Boolean, f a) -> f a
+  //.
+  //. Discards the first element which does not satisfy the predicate, and all
+  //. subsequent elements.
+  //.
+  //. This function is derived from [`concat`](#concat), [`empty`](#empty),
+  //. [`of`](#of), and [`reduce`](#reduce).
+  //.
+  //. See also [`dropWhile`](#dropWhile).
+  //.
+  //. ```javascript
+  //. > takeWhile (s => /x/.test (s), ['xy', 'xz', 'yx', 'yz', 'zx', 'zy'])
+  //. ['xy', 'xz', 'yx']
+  //.
+  //. > takeWhile (s => /y/.test (s), ['xy', 'xz', 'yx', 'yz', 'zx', 'zy'])
+  //. ['xy']
+  //.
+  //. > takeWhile (s => /z/.test (s), ['xy', 'xz', 'yx', 'yz', 'zx', 'zy'])
+  //. []
+  //. ```
+  function takeWhile(pred, foldable) {
+    //  Fast path for arrays.
+    if (Array.isArray (foldable)) {
+      var idx = 0;
+      while (idx < foldable.length && pred (foldable[idx])) idx += 1;
+      return foldable.slice (0, idx);
+    }
```

Comment:
Do you like this implementation, @gabejohnson?

Reply:
That works, but you may be able to save a traversal by just copying over elements to a new array in the loop. Same time complexity though.